### PR TITLE
TestPyPi hookups and changes for easier deployment

### PR DESCRIPTION
**Pull request by baogorek**
Besides hooking up TestPyPi to the dev branch, here are things to note:

- I made a separate "long description" for PyPi in setup.py, because our README.me file was causing the deployment to fail and I couldn't understand why.

- There's a python file in .circleci that writes .pypirc now. It was a bit more complicated with testpypi.

- This build will fail but you'll see the only thing I changed was the version back to where it was (which is why it will fail TestPyPi deployment). The build right before is a success.

- I also changed the CircleCi config to use the requirements in the base of the package instead of in the .circleci directory. I couldn't think of a good reason why we would need them in two separate places.

### CODE REVIEW

**Comment by baogorek** [comment]
Failing the build because I deployed once to TestPyPi with nothing in the README.md and it worked. I should have thought to change the version then since I knew I'd change it back. This should work though in actual PyPi.
image

File: .circleci/config.yml [hunk 1]
```diff
@@ -48,20 +48,20 @@ jobs:
       - run:
           name: init .pypirc
           command: |
-            echo -e "[pypi]" >> ~/.pypirc
-            echo -e "username = $PYPI_USERNAME" >> ~/.pypirc
-            echo -e "password = $PYPI_PASSWORD" >> ~/.pypirc
+            python .circleci/create_pypirc.py \
```
Comment (kjmerf) [inline .circleci/config.yml:51]: Cool. I guess this is necessary for the test pypi deploy?

File: .circleci/create_pypirc.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!/usr/env/python
+import sys
+
+def main(argv):
```
Comment (kjmerf) [inline .circleci/create_pypirc.py:4]: at some point we might want to make this as an argparser. But looks good for now

File: .circleci/config.yml [hunk 3]
```diff
@@ -16,27 +16,27 @@ jobs:
       # Take advantage of latest cache of dependencies
       - restore_cache:
           keys:
-            - v1-dependencies-{{ checksum ".circleci/requirements.txt" }}
+            - v1-dependencies-{{ checksum "requirements.txt" }}
```
Comment (kjmerf) [inline .circleci/config.yml:19]: what happened here? Do we just have one requirements.txt file now? I thought we intentionally had two: one for the scroller and one for circle ci... Although I never did quite understand why we needed pandas sql in both. Thoughts?

**Review by kjmerf — APPROVED** [review]

**Comment by kjmerf** [comment]
I have a couple questions but don't let that stop you. Looks good to me. Really creative with the python file to create pypirc! Can we actually merge this with the failed tests? If so, should we going forward?

**Comment by kjmerf** [comment]
FYI that I logged issue #42 to track this work.

File: .circleci/config.yml (hunk 1 above)
Comment (baogorek) [reply .circleci/config.yml:51]: I was getting auth errors for TestPyPi, so I looked up examples of .pypirc files that featured TestPyPi deployment and they were more involved. I could still write all the lines here but after a point I figured it'd be easier in a template file.

File: .circleci/create_pypirc.py (hunk 2 above)
Comment (baogorek) [reply .circleci/create_pypirc.py:4]: Yeah and if you know offhand of and easier templating mechanism let me know. I used to have a trick using just bash but I forgot it.

File: .circleci/config.yml (hunk 3 above)
Comment (baogorek) [reply .circleci/config.yml:19]: My guess is that pandasql was installing pandas and numpy which and providing all the necessary dependencies for datascroller. Without pandasql there's no pandas and you would get an error. I could see us having a requirements for datascroller alone and then extra requirements for CircleCI but we'd have two install statements.

**Comment by baogorek** [comment]
> I have a couple questions but don't let that stop you. Looks good to me. Really creative with the python file to create pypirc! Can we actually merge this with the failed tests? If so, should we going forward?

Thanks @kjmerf ! I assume we can merge but should I just update the version to 1.3.1?

**Comment by baogorek** [comment]
There are a few discussion points remaining that we should still talk about. Again, I wish I had changed the version when I first took this branch out to "1.3.0rc1" or something so then I could change it back to "1.3.0" for the merge. I'm going to send this one through and cross my figures.

**Pull request merged**